zarr/util.py

```diff
@@ -312,117 +313,126 @@ def _repr_html_(self):
-        }
-
-        div.zarr-tree li {
-            border-left: 2px solid #000;
-            margin-left: 1em;
-        }
-
-        div.zarr-tree li > div {
-            padding-left: 1.3em;
-            padding-top: 0.225em;
-            padding-bottom: 0.225em;
-        }
-
-        div.zarr-tree li > div::before {
-            content: '';
-            position: absolute;
-            top: 0;
-            left: -2px;
-            bottom: 50%;
-            width: 1.2em;
-            border-left: 2px solid #000;
-            border-bottom: 2px solid #000;
-        }
-
-        div.zarr-tree > ul > li:first-child > div {
-            padding-left: 4%;
-        }
-
-        div.zarr-tree > ul > li:first-child > div::before {
-            border: 0 none transparent;
-        }
-
-        div.zarr-tree ul > li:last-child {
-            border-left: 2px solid transparent;
-        }
-    """)
-    result += "</style>\n\n"
-
-    # Insert the HTML list
-    result += """<div class="zarr-tree">\n"""
-    result += "<ul>\n"
-    result += custom_html_sublist(group, indent=indent)
-    result += "</ul>\n"
-    result += "</div>\n"
+def tree_html(group, expand, level):
+
+    result = ''
+
+    # include CSS for jstree default theme
+    css_url = '//cdnjs.cloudflare.com/ajax/libs/jstree/3.3.3/themes/default/style.min.css'
+    result += '<link rel="stylesheet" href="{}"/>'.format(css_url)
+
+    # construct the tree as HTML nested lists
+    node_id = uuid.uuid4()
+    result += '<div id="{}" class="zarr-tree">'.format(node_id)
+    result += '<ul>'
+    root = TreeNode(group, level=level)
+    result += tree_html_sublist(root, root=True, expand=expand)
+    result += '</ul>'
+    result += '</div>'
+
+    # construct javascript
+    result += dedent("""
+        <script>
+            if (!require.defined('jquery')) {
+                require.config({
+                    paths: {
+                        jquery: '//cdnjs.cloudflare.com/ajax/libs/jquery/1.12.1/jquery.min'
+                    },
+                });
+            }
+            if (!require.defined('jstree')) {
+                require.config({
+                    paths: {
+                        jstree: '//cdnjs.cloudflare.com/ajax/libs/jstree/3.3.3/jstree.min'
+                    },
+                });
+            }
+            require(['jstree'], function() {
+                $('#%s').jstree({
+                    types: {
+                        Group: {
+                            icon: "%s"
+                        },
+                        Array: {
+                            icon: "%s"
+                        }
+                    },
+                    plugins: ["types"]
+                });
+            });
+        </script>
+    """ % (node_id, tree_group_icon, tree_array_icon))
 
     return result
 
 
+tree_group_icon = 'fa fa-folder'
+tree_array_icon = 'fa fa-table'
+# alternatives...
+# tree_group_icon: 'jstree-folder'
+# tree_array_icon: 'jstree-file'
```

Comment:
Maybe these should be variables in `TreeViewer`.

Reply:
Yes could be. I was trying to think of best way to make it so a really enthusiastic user could change the default icons if they want to use their favourite emojis instead :-)

Reply:
Was thinking about having static variables of the class as an option as well. Maybe that would work better?